src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -1527,6 +1533,9 @@ SyncMetadataToNodes(void)
 
 			if (!SyncMetadataSnapshotToNode(workerNode, raiseInterrupts))
 			{
+				ereport(WARNING, (errmsg("failed to sync metadata to %s:%d",
+										 workerNode->workerName,
+										 workerNode->workerPort)));
```

Comment:
do we hit this warning in case of a deadlock?

Reply:
If the daemon itself is cancelled because of distributed deadlock, we should get  `ERROR: canceling statement due to user request`, which will cause the daemon to `exit(1)` (in elog.c's handlers), and restarted in few seconds.

If another backend which is blocking the metadata_sync is cancelled, `SyncMetadataSnapshotToNode` should be able to succeed as it got all required locks.

This WARNING can happen in situations like connection problems or when workerNode is read-only.